### PR TITLE
Clean up of batch framework and mc batch docs

### LINKED ISSUE #748: [BUG] Minor fixups for batch replication
Need to fix the following areas in batch replication docs:

- [x] Remove reference to [lifecycle management](https://min.io/docs/minio/linux/reference/minio-mc/mc-batch-generate.html#mc.batch.generate.TARGET) from `mc batch` subcommands, if any
- [x] The YAML `source.type` and `target.type` must be ``minio`` - Found [here](https://min.io/docs/minio/linux/reference/minio-mc/mc-batch-generate.html#replicate), source code [here](https://github.com/minio/minio/blob/0c1f8b4e0f044b20102268872b5dc3d201c410fb/cmd/batch-handlers.go#L180)
- [x] The indentation on the YAML appears to have gone out of sorts
- [x] The # NOTE: blocks in the YAML make reference for remote -> local, which is not yet supported. this also requires an upstream change to [madmin-go](https://github.com/minio/madmin-go/blob/main/batch-job.go#L48-L101)

After internal discussion we also noted some confusion around the usage of TARGET, and we need to clarify support for local -> remote and remote -> local. A few other tidy-ups might come out of that.

### PR DESCRIPTION
**Pull request by djwfyi**
- Removes bad copy paste info about ilm from batch command docs.
- Clarifies that batch files are one task only.
- Fixes source type from `s3` to `minio`.
- Cleans up indentation errors in replicate yaml example.
- Replicate task is local to remote only.

Closes #748

### CODE REVIEW

File: source/administration/batch-framework.rst
Comment (ravindk89) [inline source/administration/batch-framework.rst:87]:
```suggestion
The ``replicate`` job type performs a single-run replication "push" of objects from the local MinIO deployment to another MinIO deployment.
```

Comment (ravindk89) [reply source/administration/batch-framework.rst:87]: something about "push replication" reads odd to me.

File: source/reference/minio-mc/mc-batch-generate.rst
Comment (ravindk89) [inline source/reference/minio-mc/mc-batch-generate.rst:249]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude

Might be helpful for single-sourcing the sample YAML?

**Review by ravindk89 — APPROVED** [review]
LGTM mod nits, no need for second review if build passes.

File: source/reference/minio-mc/mc-batch-generate.rst
Comment (djwfyi) [reply source/reference/minio-mc/mc-batch-generate.rst:249]: 👍

File: source/reference/minio-mc/mc-batch-generate.rst
Comment (djwfyi) [reply source/reference/minio-mc/mc-batch-generate.rst:131]: 👍

**Pull request merged**
